components/coins-status/index.js

```diff
@@ -1,39 +1,69 @@
-import { useEffect } from 'react';
+import { useState, useEffect } from 'react';
 import { useAnimateValue } from './useCoinStatus';
 import styles from './coin.module.css';
 import PropTypes from 'prop-types';
 
+const BASE_API_URL = process.env.NEXT_PUBLIC_BASE_API_URL;
+const WALLET_URL = `${BASE_API_URL}/wallet`;
+
 const Coins = (props) => {
-  const [coins, changeCoins] = useAnimateValue(0, props.coin.value, 5000);
+  const [coins, changeCoins] = useAnimateValue(0, props.balance, 5000);
+
   useEffect(() => {
-    changeCoins(props.coin.value);
-  }, [props.coin.value, changeCoins]);
+    changeCoins(props.balance);
+  }, [props.balance, changeCoins]);
+
   return (
     <div className={styles.cointypeIndicator}>
       <div className={styles.coinData}>
-        <p> {props.coin.name}</p>
-        <p> {coins} </p>
+        <p> {props.coin}</p>
+        <p> {props.balance} </p>
       </div>
       <div className={`coin ${styles.rotateVertCenter}`}></div>
       <style jsx>{`
         .coin {
-          background-color: ${props.coin.color};
-          width: calc(4.25em / 2);
-          height: calc(4.25em / 2);
+          background-color: ${props.coin === 'neelam'
+            ? 'blue'
+            : props.coin === 'dinero'
+            ? 'green'
+            : 'gray'};
+          width: calc(5em / 2);
+          height: calc(5em / 2);
           border-radius: 50%;
           box-shadow: 3px 3px 3px #000;
-          border: 1px solid ${props.coin.borderColor};
+          border: 1px solid
+            ${props.coin === 'neelam'
+              ? 'blue'
+              : props.coin === 'dinero'
+              ? 'green'
+              : 'gray'};
         }
       `}</style>
     </div>
   );
 };
 
 const CoinsStatus = (props) => {
-  let coinsArray = props.coins.map((coin, index) => {
-    return <Coins coin={coin} key={index} />;
-  });
-  return <div className={styles.coinsContainer}>{coinsArray}</div>;
+  const [currencies, setCurrencies] = useState();
+
+  useEffect(async () => {
+    let walletJson,
+      walletData,
+      currencyBalance,
+      coinsArray = [];
+
+    walletData = await fetch(WALLET_URL, { credentials: 'include' });
+    walletJson = await walletData.text();
```

Comment:
I am confused about the usage of .text() here , I used to think about .json at this point

Reply:
Thanks for pointing out. I have modified code to use more standard .json() method